machinator-scala/src/Machinator/Scala/Scheme/Types/Codegen.hs

```diff
@@ -75,11 +86,35 @@ genTypeV1 ty =
       string "List" <> WL.brackets (genTypeV1 t2)
     MaybeT t2 ->
       string "Option" <> WL.brackets (genTypeV1 t2)
+    MapT k v ->
+      string "Map" <> WL.brackets (genTypeV1 k <> "," <+> genTypeV1 v)
+
+-- | Find the set of common properties in the constructors of a variant.
+variantProperties :: [(Name, Maybe Docs, [(Name, Type)])] -> [(Name, Type)]
+variantProperties ctors =
+  let fs = fmap (\(_, _, f) -> S.fromList f) ctors
+  in S.toAscList (foldr S.intersection (S.unions fs) fs)
```

Comment:
If this accepts the non-empty list of constructors and then `foldl1 S.intersection` without the extra `S.unions` is total and clearer IMO.